File: base/049/Main.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Locale;

public class Main {
    public static void main(String[] a) {
        Locale.setDefault(new Locale("en", "US"));
        ArrayList<String> vet = new ArrayList<>();
        
        while (true) {
            var line = input();
            write("$" + line);
            var args = line.split(" ");

            if (args[0].equals("end")) {
                break;
            }
            else if(args[0].equals("push")) {
                for (int i = 1; i < args.length; i++) {
                    vet.add(args[i]);
                }
            }
            else if(args[0].equals("show")) {
                write("[" + String.join(", ", vet) + "]");
            }
            else if(args[0].equals("erase")) {
                vet.remove((int) number(args[1]));
            }
            else {
                write("fail: comando invalido");
            }
        }
    }

    private static Scanner scanner = new Scanner(System.in);
    private static String  input()              { return scanner.nextLine(); }
    private static double  number(String value) { return Double.parseDouble(value); }
    private static void    write(String value)  { System.out.println(value); }
}

File: base/049/main.ts
let _cin_: string[] = require("fs").readFileSync(0).toString().split("\n");
let input = () : string => _cin_.length === 0 ? "" : _cin_.shift()!;
let write = (text: any, end:string="\n")=> process.stdout.write("" + text + end);

function main() {
    let vet = new Array<string>();

    while (true) {
        let line = input();
        write("$" + line);
        let args = line.split(" ");

        if      (args[0] === "end")   { break;                                                       }
        else if (args[0] === "push")  { 
            for (let i = 1; i < args.length; i++) { 
                vet.push(args[i]); 
            } 
        }
        else if (args[0] === "show")  { write ("[" + vet.join(", ") + "]");                          }
        else if (args[0] === "erase") { vet.splice(+args[1], 1);                                     }
        else                          { write("fail: comando invalido");                             }
    }
}

main();